All apps as json

henrikplate opened the pull request:
The changes allow to specify the desired format when exporting all apps using the `HTTP GET apps/csv?format=[csv|json]` endpoint. Admittedly, the name is misleading, but it is kept for backward compatibility.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/vulnerability-assessment-tool?pullRequest=11) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/vulnerability-assessment-tool?pullRequest=11) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/vulnerability-assessment-tool?pullRequest=11) it.</sub>

Naramsim commented:
Why not creating also an endpoint `apps/json`? Where you actually have the parameter inside the path?

henrikplate commented:
@Naramsim : Would be an option, I guess it is a matter of taste. In the long run, I wanted to change the endpoint to `apps/export?format=[json|csv]`, but I kept it in order to not cause troubles for API users.

serenaponta commented:
The code changes look fine to me but when testing it:
1) the test case is failing 
`Failed tests:
  ApplicationControllerTest.testExportApp:216 Status expected:<200> but was:<404>
Tests in error:
  ArtifactMapsTest.testsBuildMaps:25 NullPointer
`
I also tested the packaged version and 
- json via mail worked successfully
- json as response `There was an unexpected error (type=Internal Server Error, status=500).
IOError writing file to output stream`
- csv to be tested

I the console output of the backend I observed several of the following: 

`2018-10-18 16:25:13.323 ERROR 19800 --- [pool-6-thread-2] c.s.p.v.b.c.ApplicationExporterThread    : [java.lang.ArrayIndexOutOfBoundsException] while appending data for app [public:XXX:XXX:XXX], entry [SPACE-ID;public;[];;XXX;XXX;XXX;XXX;;] will not be appended to [JSON]: null
`

henrikplate commented:
@Serena:
I fixed testExportApp, the other error seems unrelated to my modification
I also fixed the JSON format (closing array bracket was missing) and re-arranged space properties
I could not deploy the WAR to vulas-dev 40001 or 40002, the startup failed (but I did not have the time to investigate)
Feel free to have a look, but do not spend much time to make the deployment on vulas-dev work, the feature is not that important...

serenaponta commented:
testExportApp passes. 

I deployed the backend in tomcat1, however the runtime is very slow. I think this may be due to the fact that some indexes "get lost" when we restore the db from the dump  (i observed it in the past).

As a result i didn't check the API, exp. I never run it **without** email

henrikplate commented:
Renamed endpoint from `/apps/csv` to `/apps/export`
Tested all combinations of `format` (json|csv) and `to` (with and without)
Remaining: Check whether/why space properties are not shown on prod system
